src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/TranscriptionIdControllerAuthorisationImpl.java

```diff
@@ -32,18 +32,14 @@ public ContextIdEnum getContextId() {
         return TRANSCRIPTION_ID;
     }
 
+    public String getEntityIdParam() {
+        return TRANSCRIPTION_ID_PARAM;
+    }
+
     @Override
     public void checkAuthorisation(HttpServletRequest request, Set<SecurityRoleEnum> roles) {
-        Optional<String> transcriptionIdParamOptional = getPathParamValue(request, TRANSCRIPTION_ID_PARAM);
-        checkAuthorisationByTranscriptionId(transcriptionIdParamOptional, roles);
-
-        if (transcriptionIdParamOptional.isEmpty()) {
-            transcriptionIdParamOptional = Optional.ofNullable(request.getParameter(TRANSCRIPTION_ID_PARAM));
-            checkAuthorisationByTranscriptionId(transcriptionIdParamOptional, roles);
-        }
-
-        if (transcriptionIdParamOptional.isEmpty()) {
-            transcriptionIdParamOptional = Optional.ofNullable(request.getHeader(TRANSCRIPTION_ID_PARAM));
+        Optional<String> transcriptionIdParamOptional = getEntityParamOptional(request, TRANSCRIPTION_ID_PARAM);
+        if (transcriptionIdParamOptional.isPresent()) {
```

Comment:
No need to have this isPresent line duplicated here if we have it also in the method checkAuthorisationByTranscriptionId line #63?